### PR TITLE
(CDAP-4221) Not to call getHadoopConf() from BatchContext

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cassandra-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchCassandraSource.java [hunk 1]
```diff
@@ -111,7 +114,7 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     }
     CqlConfigHelper.setInputCql(conf, config.query);
 
-    job.setInputFormatClass(CqlInputFormat.class);
+    context.setInput(Input.of("Cassandra", new SourceInputFormatProvider(CqlInputFormat.class, conf)));
```
Comment (gokulavasan) [inline cassandra-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchCassandraSource.java]: Hmm, I made and merged this change in my PR https://github.com/caskdata/hydrator-plugins/pull/235. Not sure why it is not showing up in the diff?

Comment (gokulavasan) [inline cassandra-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchCassandraSource.java]: @chtyim Can you rebase your changes on top of the latest release/1.3 ?

Comment (gokulavasan) [inline cassandra-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchCassandraSource.java]: Ok nvmind, it shows a conflict now.

File: hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/HiveBatchSink.java [hunk 2]
```diff
@@ -91,7 +93,19 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
 
   @Override
   public void prepareRun(BatchSinkContext context) throws Exception {
-    Job job = context.getHadoopJob();
+    Job job = JobUtils.createInstance();
+    Configuration conf = job.getConfiguration();
+
+    if (UserGroupInformation.isSecurityEnabled()) {
```
Comment (gokulavasan) [inline hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/HiveBatchSink.java]: Looks like JobUtils.createInstance already has this logic?

File: hydrator-common/src/main/java/co/cask/hydrator/common/batch/JobUtils.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  /**
+   * Creates a new instance of {@link Job}. Note that the job created is not meant for actual MR
+   * submission. It's just for setting up configurations.
+   */
+  public static Job createInstance() throws IOException {
+    Job job = Job.getInstance();
+    Configuration conf = job.getConfiguration();
+    conf.clear();
+
+    if (UserGroupInformation.isSecurityEnabled()) {
```
Comment (gokulavasan) [inline hydrator-common/src/main/java/co/cask/hydrator/common/batch/JobUtils.java:40]: Should all sources/sinks use this instance to create Job/Configuration? In this PR, some have been changed to use this and some don't ?

**Comment by gokulavasan** [comment]
Couple of questions. Will review again once they are addressed.

File: hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/HiveBatchSink.java (hunk 2 above)
Comment (chtyim) [inline hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/HiveBatchSink.java]: That's right.

File: hydrator-common/src/main/java/co/cask/hydrator/common/batch/JobUtils.java (hunk 3 above)
Comment (chtyim) [inline hydrator-common/src/main/java/co/cask/hydrator/common/batch/JobUtils.java:40]: I think everyone who needs a `Job` already using this method. There are some cases that `Job` is not needed, but instead only need `Configuration`. For those cases, they are not calling this method.

Comment (gokulavasan) [inline hydrator-common/src/main/java/co/cask/hydrator/common/batch/JobUtils.java:40]: Oh ok sounds good.

File: hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/HiveBatchSink.java (hunk 2 above)
Comment (gokulavasan) [inline hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/HiveBatchSink.java]: So will this be removed?

**Comment by gokulavasan** [comment]
LGTM 👍

**Comment by chtyim** [comment]
Sqaushed and waiting for build.

**Pull request merged**